src/main/java/edu/usf/cutr/grha/otp/OtpService.kt

```diff
@@ -0,0 +1,105 @@
+package edu.usf.cutr.grha.otp
+
+import edu.usf.cutr.grha.model.ChicagoTncData
+import edu.usf.cutr.grha.utils.GtfsUtils
+import edu.usf.cutr.otp.plan.api.PlanApi
+import edu.usf.cutr.otp.plan.model.Planner
+import edu.usf.cutr.otp.plan.model.RequestParameters
+import io.reactivex.Observable
+import io.reactivex.Single
+import io.reactivex.SingleOnSubscribe
+import kotlinx.coroutines.*
+import kotlinx.coroutines.Dispatchers.Main
+import kotlinx.coroutines.flow.*
+import kotlin.coroutines.CoroutineContext
+import kotlin.coroutines.resumeWithException
+
+class OtpService(
+    private var chicagoTncData: List<ChicagoTncData>,
+    private var url: String
+) {
+
+    fun call() {
+        makePlanFlow()
+    }
+
+    private fun getPlanData() {
+        var i = 0
+        var start = 0L
+        val planFlow: Flow<Planner> = flow {
+            while (i < 42) {
+                if (chicagoTncData[i].pickupCentroidLatitude == 0.0 ||
```

Comment:
I'd put this test for valid locations in a private `validLocation()` method to help with readability.